tests/compiler.js

```diff
@@ -485,6 +485,27 @@
                   { name: 'james' },
                   'FooInclude james');
 
+            equal('{% macro foo() %}{% set x = "foo"%}{{ x }}{% endmacro %}' +
+                  '{% set x = "bar" %}' +
+                  '{{ x }}' +
+                  '{{ foo() }}' +
+                  '{{ x }}',
+                  'barfoobar');
+
+            // this test checks that the issue #577 is resolved
+            equal('{% macro foo(topLevel = "false", prefix = "") %}' +
```

Comment:
Using the string "false" as the default parameter here seems needlessly misleading (since the string "false" is actually truthy). Either eliminate the default (since you aren't actually relying on it anywhere, or make it just `false` instead of `"false"`. Also, no spaces around the `=` in parameter lists.